packages/api/test/src/domain/bufficorn.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { Bufficorn } from '../../../src/domain/bufficorn'
+import { Trait, BufficornLeaderboardInfo } from '../../../src/types'
+
+describe('bufficorn.ts', () => {
+  it('getScore', async () => {
+    const bufficorn = new Bufficorn(undefined, 0)
+    expect(bufficorn.calculateScore()).toStrictEqual(0)
+    expect(bufficorn.calculateScore(Trait.Vigor)).toStrictEqual(0)
+
+    bufficorn.stats.vigor = 10
+    expect(bufficorn.calculateScore()).toStrictEqual(0)
+    expect(bufficorn.calculateScore(Trait.Vigor)).toStrictEqual(10)
+
+    bufficorn.stats.agility = 20
+    bufficorn.stats.coat = 30
+    bufficorn.stats.coolness = 40
+    bufficorn.stats.speed = 50
+    bufficorn.stats.stamina = 60
+    expect(bufficorn.calculateScore()).toStrictEqual(10)
+    expect(bufficorn.calculateScore(Trait.Vigor)).toStrictEqual(10)
+    expect(bufficorn.calculateScore(Trait.Agility)).toStrictEqual(20)
+    expect(bufficorn.calculateScore(Trait.Coat)).toStrictEqual(30)
+    expect(bufficorn.calculateScore(Trait.Coolness)).toStrictEqual(40)
+    expect(bufficorn.calculateScore(Trait.Speed)).toStrictEqual(50)
+    expect(bufficorn.calculateScore(Trait.Stamina)).toStrictEqual(60)
+  })
```

Comment:
Personal preference: I understand better the test when they follow [Given-When-Then](https://en.wikipedia.org/wiki/Given-When-Then) structure

```suggestion
  it('getScore', async () => {
    const bufficorn = new Bufficorn(undefined, 0)

    bufficorn.stats.vigor = 10
    bufficorn.stats.agility = 20
    bufficorn.stats.coat = 30
    bufficorn.stats.coolness = 40
    bufficorn.stats.speed = 50
    bufficorn.stats.stamina = 60
    
    expect(bufficorn.calculateScore()).toStrictEqual(0)
    expect(bufficorn.calculateScore(Trait.Vigor)).toStrictEqual(10)
    expect(bufficorn.calculateScore()).toStrictEqual(0)
    expect(bufficorn.calculateScore(Trait.Vigor)).toStrictEqual(0)
    expect(bufficorn.calculateScore()).toStrictEqual(10)
    expect(bufficorn.calculateScore(Trait.Vigor)).toStrictEqual(10)
    expect(bufficorn.calculateScore(Trait.Agility)).toStrictEqual(20)
    expect(bufficorn.calculateScore(Trait.Coat)).toStrictEqual(30)
    expect(bufficorn.calculateScore(Trait.Coolness)).toStrictEqual(40)
    expect(bufficorn.calculateScore(Trait.Speed)).toStrictEqual(50)
    expect(bufficorn.calculateScore(Trait.Stamina)).toStrictEqual(60)
  })
```